### PR TITLE
SRDF generator now loads multi-robot move group states from yaml

### PR DESCRIPTION
**Pull request by ethanfowler**
## Proposed changes

SRDF generator now loads multi-robot move group states from a yaml file. These group states can reference each other or SRDF-defined groups. Adding "relaxed hand" states and including them in arm start states (only if hand move groups exist).

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [x] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [ ] Manually tested on hardware (if hardware specific or related).

## Documentation

- [ ] No documentation required to be added.
- [ ] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [x] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

**Review by maxzieba — APPROVED** [review]

File: sr_multi_moveit/sr_multi_moveit_config/scripts/generate_robot_srdf.py [hunk 1]
```diff
@@ -394,50 +407,94 @@ def parse_arm_groups(self, manipulator_id, manipulator):
...
+                    group_states[group][state_name][group_state_child.getAttribute("name")] = \
+                        group_state_child.getAttribute("value")
+        return group_states
+
+    # Generate multi-robot move group states
+    def add_move_group_states(self):
+        # Check which move groups are in the generated SRDF
+        self.new_robot_srdf.seek(0)
+        move_group_names = list(set(re.findall(r'<group\s+.*name="([^"]*)"', self.new_robot_srdf.read())))
+        # Collect the move group states defined in the seperate robot SRDFs
```
Comment (toliver) [inline sr_multi_moveit/sr_multi_moveit_config/scripts/generate_robot_srdf.py]: typo "separate"

File: sr_multi_moveit/sr_multi_moveit_config/scripts/generate_robot_srdf.py [hunk 2]
```diff
@@ -394,50 +407,94 @@ def parse_arm_groups(self, manipulator_id, manipulator):
...
+                        if not self.create_move_group_state(ancestor_group, ancestor_group_state):
+                            return False
+                # The ancestor group state either already existed or has been created
+                ancestor_move_group_state = self._move_group_states[ancestor_group][ancestor_group_state]
+                for joint, value in ancestor_move_group_state.items():
+                    if joint not in combined_move_group_state["joint_angles"]:
+                        combined_move_group_state["joint_angles"][joint] = value
+        if group_name not in self._move_group_states:
+            self._move_group_states[group_name] = {}
+        self._move_group_states[group_name][group_state_name] = combined_move_group_state["joint_angles"]
```
Comment (toliver) [inline sr_multi_moveit/sr_multi_moveit_config/scripts/generate_robot_srdf.py]: I'm lacking a bit of context about the whole information flow of srdfs and yamls into these data structures.
Not sure what do we do with `_move_group_states` and `_new_move_group_states`.
Is there a document describing a bit this flow? If not, would it be a good time to write an initial draft, if that's clear in your head right now?

**Review by toliver — CHANGES_REQUESTED** [review]

File: sr_multi_moveit/sr_multi_moveit_config/scripts/generate_robot_srdf.py (hunk 2 above)
Comment (ethanfowler) [reply sr_multi_moveit/sr_multi_moveit_config/scripts/generate_robot_srdf.py]: Added some documentation and renamed some variables to make things clearer in https://github.com/shadow-robot/sr_interface/pull/768/commits/c17efa1cd76b3ae46e55204f9b9d8b4b765f4c4c.

File: sr_multi_moveit/sr_multi_moveit_config/scripts/generate_robot_srdf.py (hunk 1 above)
Comment (ethanfowler) [reply sr_multi_moveit/sr_multi_moveit_config/scripts/generate_robot_srdf.py]: Fixed in https://github.com/shadow-robot/sr_interface/pull/768/commits/c17efa1cd76b3ae46e55204f9b9d8b4b765f4c4c.

**Comment by ethanfowler** [comment]
Ready for re-review @toliver !

File: sr_multi_moveit/sr_multi_moveit_config/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# sr_multi_moveit_config
+
+Configuration of multiple-robots running as a combined robot.
+
+## Multi-Robot Move Groups and States
+
+At run time, [generate_robot_srdf.py](scripts/generate_robot_srdf.py) parses individual robot (UR arms, Shadow Hands) U/SRDFs and creates move groups that include links from multiple robot definitions. It also loads and parses a configuration file ([multi_robot_move_group_states.yaml](config/multi_robot_move_group_states.yaml) by default) that defines named joint states for these combined robot move groups. These combined move group states can inherit joint angles from states defined in individual robot U/SRDFs.
```
Comment (toliver) [inline sr_multi_moveit/sr_multi_moveit_config/README.md:7]: The only thing i'd like to see added to this is what is the output of `generate_robot_srdf.py` after parsing all this information.
Is it a new robot_semantic_description loaded to param server?

**Review by toliver — CHANGES_REQUESTED** [review]

File: sr_multi_moveit/sr_multi_moveit_config/README.md (hunk 3 above)
Comment (ethanfowler) [reply sr_multi_moveit/sr_multi_moveit_config/README.md:7]: Added a conclusion in https://github.com/shadow-robot/sr_interface/pull/768/commits/b59c57698fc0c4ea7c7c6dbfaff00c4e3d4fa287.

Comment (toliver) [reply sr_multi_moveit/sr_multi_moveit_config/README.md:7]: Should it be `/robot_semantic_description` ?

Comment (ethanfowler) [reply sr_multi_moveit/sr_multi_moveit_config/README.md:7]: Whoops fixed in https://github.com/shadow-robot/sr_interface/pull/768/commits/87510eed6e1d93adfd7947b648956e6c00d8db70.

**Review by toliver — APPROVED** [review]

**Pull request merged**